src/Symfony/Component/DependencyInjection/Compiler/FactoryReturnTypePass.php

```diff
@@ -1,108 +0,0 @@
-<?php
-
-/*
- * This file is part of the Symfony package.
- *
- * (c) Fabien Potencier <fabien@symfony.com>
- *
- * For the full copyright and license information, please view the LICENSE
- * file that was distributed with this source code.
- */
-
-namespace Symfony\Component\DependencyInjection\Compiler;
-
-use Symfony\Component\DependencyInjection\ContainerBuilder;
-use Symfony\Component\DependencyInjection\Definition;
-use Symfony\Component\DependencyInjection\Reference;
-
-/**
- * @author Guilhem N. <egetick@gmail.com>
- *
- * @deprecated since version 3.3, to be removed in 4.0.
- */
-class FactoryReturnTypePass implements CompilerPassInterface
-{
-    private $resolveClassPass;
-
-    public function __construct(ResolveClassPass $resolveClassPass = null)
-    {
-        if (null === $resolveClassPass) {
-            @trigger_error('The '.__CLASS__.' class is deprecated since version 3.3 and will be removed in 4.0.', E_USER_DEPRECATED);
-        }
-        $this->resolveClassPass = $resolveClassPass;
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function process(ContainerBuilder $container)
-    {
-        $resolveClassPassChanges = null !== $this->resolveClassPass ? $this->resolveClassPass->getChanges() : array();
```

Comment:
do we still need the `ResolveClassPass::getChanges()` method (which requires making the compiler pass stateful) ?